contracts/DCAStrategies/DCAStrategies/DCAStrategiesManagementHandler.sol

```diff
@@ -4,18 +4,45 @@ pragma solidity >=0.8.7 <0.9.0;
 import '../../interfaces/IDCAStrategies.sol';
 
 abstract contract DCAStrategiesManagementHandler is IDCAStrategiesManagementHandler {
+  mapping(uint80 => Strategy) internal _strategies;
+  mapping(string => uint80) internal _strategyNames;
+  mapping(bytes32 => IDCAStrategies.ShareOfToken) internal _tokenShares;
+  uint80 public strategyCounter; // NOTE: could we use _mintCounter from DCAStrategiesPermissionsHandler?
+
+  function _checkTokenSharesSanity(IDCAStrategies.ShareOfToken memory _tokens) internal pure returns (bool) {
+    // TODO
+    return true;
+  }
+
+  function _getTokenSharesKey(uint256 _strategyId, uint80 _version) internal pure returns (bytes32) {
+    // NOTE:
+    return keccak256(abi.encodePacked(_strategyId, _version));
+  }
+
   /// @inheritdoc IDCAStrategiesManagementHandler
-  function getStrategy(uint80 _strategyId) external view override returns (IDCAStrategies.Strategy memory) {}
+  function getStrategy(uint80 _strategyId) external view virtual override returns (IDCAStrategies.Strategy memory) {}
 
   /// @inheritdoc IDCAStrategiesManagementHandler
-  function getStrategyIdByName(string memory _strategyName) external view override returns (uint80 _strategyId) {}
+  function getStrategyIdByName(string memory _strategyName) external view virtual override returns (uint80 _strategyId) {}
 
   /// @inheritdoc IDCAStrategiesManagementHandler
   function createStrategy(
     string memory _strategyName,
-    IDCAStrategies.ShareOfToken[] memory _tokens,
+    IDCAStrategies.ShareOfToken memory _tokens,
     address _owner
-  ) external override returns (uint80 _strategyId) {}
+  ) external override returns (uint80 _strategyId) {
+    if (_owner == address(0)) revert IDCAStrategies.ZeroAddress(); // NOTE: should users be able to assign ownership to any address? what if they use it for scamming? e.g: give ownership of a strategy to vitalik.eth and use it to trick users into buying shitcoins
+    if (_strategyNames[_strategyName] != 0) revert NameAlreadyExists();
```

Comment:
Let's also check that the name is <= 32 bytes. I think you can do something like `bytes(_strategyName).length`